src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -6396,27 +6396,27 @@ public static void Main()
 }";
             var compilation = CreateCompilation(source, options: TestOptions.DebugExe);
             compilation.VerifyDiagnostics(
-                // (6,18): error CS8405: A default literal 'default' is not valid as a pattern. Use another literal (e.g. '0' or 'null') as appropriate. To match everything, use a discard pattern '_'.
+                // (6,18): error CS8716: No target type was found for the default literal.
                 //         if (i is default) {} // error 1
-                Diagnostic(ErrorCode.ERR_DefaultPattern, "default").WithLocation(6, 18),
-                // (7,19): error CS8405: A default literal 'default' is not valid as a pattern. Use another literal (e.g. '0' or 'null') as appropriate. To match everything, use a discard pattern '_'.
+                Diagnostic(ErrorCode.ERR_DefaultLiteralNoTargetType, "default").WithLocation(6, 18),
+                // (7,19): error CS8716: No target type was found for the default literal.
                 //         if (i is (default)) {} // error 2
-                Diagnostic(ErrorCode.ERR_DefaultPattern, "default").WithLocation(7, 19),
-                // (8,21): error CS8405: A default literal 'default' is not valid as a pattern. Use another literal (e.g. '0' or 'null') as appropriate. To match everything, use a discard pattern '_'.
+                Diagnostic(ErrorCode.ERR_DefaultLiteralNoTargetType, "default").WithLocation(7, 19),
+                // (8,21): error CS8716: No target type was found for the default literal.
                 //         if (i is (((default)))) {} // error 3
-                Diagnostic(ErrorCode.ERR_DefaultPattern, "default").WithLocation(8, 21),
-                // (9,27): error CS8313: A default literal 'default' is not valid as a case constant. Use another literal (e.g. '0' or 'null') as appropriate. If you intended to write the default label, use 'default:' without 'case'.
+                Diagnostic(ErrorCode.ERR_DefaultLiteralNoTargetType, "default").WithLocation(8, 21),
+                // (9,27): error CS8716: No target type was found for the default literal.
                 //         switch (i) { case default: break; } // error 4
-                Diagnostic(ErrorCode.ERR_DefaultInSwitch, "default").WithLocation(9, 27),
-                // (10,28): error CS8313: A default literal 'default' is not valid as a case constant. Use another literal (e.g. '0' or 'null') as appropriate. If you intended to write the default label, use 'default:' without 'case'.
```

Comment:
It seems too bad to lose this error message. People may have to blink a few more times after writing `case default:` before realizing how to fix their code.

Reply:
I agree. We're trading off more generalized handling of target typing, but loosing some specificity of diagnostics.

---
In reply to: [314092323](https://github.com/dotnet/roslyn/pull/37596#discussion_r314092323) [](ancestors = 314092323)